src/models/cost_function_timeseries.jl

```diff
@@ -0,0 +1,112 @@
+function get_variable_cost(
+    ts::IS.TimeSeriesData,
+    ::Union{MarketBidCost, ReserveDemandCurve},
+)
+    data = TimeSeries.TimeArray(TimeSeries.timestamp(ts), map(VariableCost, ts.data))
+    return data
+end
+
+function get_variable_cost(
+    device::StaticInjection,
+    cost::Union{MarketBidCost, ReserveDemandCurve};
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    time_series_key = get_variable(cost)
+    raw_data = IS.get_time_series_values(
+        time_series_key.time_series_type,
+        device,
+        time_series_key.name,
+        start_time = start_time,
+        len = len,
+    )
+    time_stamps = IS.get_time_series_timestamps(
+        time_series_key.time_series_type,
+        device,
+        time_series_key.name,
+        start_time = start_time,
+        len = len,
+    )
+    return TimeSeries.TimeArray(time_stamps, map(VariableCost, raw_data))
+end
+
+function get_services_bid(
+    device::StaticInjection,
+    cost::MarketBidCost,
+    service::Service;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    services_dict = get_ancillary_services(cost)
+    time_series_type = services_dict[get_name(service)]
```

Comment:
`ancillary_services` stores instances of `IS.TimeSeriesMetadata`, so this doesn't look correct.  Is that intentional?